pandas/core/dtypes/dtypes.py

```diff
@@ -375,12 +375,28 @@ def __eq__(self, other: Any) -> bool:
             # but same order is not necessary.  There is no distinction between
             # ordered=False and ordered=None: CDT(., False) and CDT(., None)
             # will be equal if they have the same categories.
-            if (
-                self.categories.dtype == other.categories.dtype
-                and self.categories.equals(other.categories)
-            ):
+            left = self.categories
+            right = other.categories
+            if not left.dtype == right.dtype:
```

Comment:
is_dtype_equal

Reply:
thatd require a circular import (actually id like to put is_dtype_equal "above" this file, but thats for another day)